htsget.md

```diff
@@ -280,6 +287,12 @@ _optional object_
 </td><td>
 For HTTPS URLs, the server may supply a JSON object containing one or more string key-value pairs which the client MUST supply as headers with any request to the URL. For example, if headers is `{"Range": "bytes=0-1023", "Authorization": "Bearer xxxx"}`, then the client must supply the headers `Range: bytes=0-1023` and `Authorization: Bearer xxxx` with the HTTPS request to the URL.
 </td></tr>
+<tr markdown="block"><td>
+`class`
+_string_
+</td><td>
+For file formats whose specification describes a header and a body, the class indicates which of the two will be retrieved when querying this URL. Either all or none of the URLs in the response must have a class attribute. The allowed values are `header` and `body`.
```

Comment:
For explicitness, I'd add something like "if class attributes are absent, client should assume data blocks include both header and body, possibly mixed"

Reply:
I don't think this is a good idea. For instance, the EVA server provides separate URLs for headers and body by default, to facilitate downloading them separately even when the `class` field isn't available.

Reply:
But if the server doesn't annotate the URLs with `class` fields, there is no way for the client to know that the EVA server is doing that. The text @mlin suggests just makes explicit the reality that clients can't make assumptions when `class` fields are missing.

Reply:
Maybe I misunderstood what @mlin means. Which of these 2 is it?

1) Each URL could contain header or body
2) Each URL could contain header and body

If it is the latter, then a client would have to split the contents from each URL in order to build a valid output file.

Reply:
If `class` fields are absent, the client can make no assumptions about the contents of individual ticket URLs or the boundaries between their contents: each URL could contain headers, body records, partial headers or records, or both.¹

(Consider e.g. a request with no referenceName/start/end thus a request for the entire file, and the server just returns a ticket that chops it up into 1 Mb chunks.)

It's implicit that the client proceeds _as if_ it's concatenating the contents of all the ticket URLs in order, to get the full file contents. It might choose to avoid redownloading an URL or two because it's already got it (effectively) cached, but that's its business. I don't quite see what you're getting at about the “client [having] to split the contents”…?

----
¹ Actually perhaps some implementations would like data records or compression blocks not to be split across ticket URLs, but at present I don't think the protocol says anything about that — so such splitting is allowed.

Reply:
Thanks for the explanation, I hadn't considered how they could be mixed due to the fix-sized chunks :sweat_smile:  Now it's all clear.

I thought it meant that **all** the URLs could contain header and body, which would make individual blocks correct, but not the response as a whole.

Reply:
Oh I see :smile: — yes: the client can't assume anything about the individual URL contents, but it **is** safe in expecting that the whole concatenated response is a valid header-body-body-…-body-EOFtrailer stream.